server/src/main/java/org/elasticsearch/action/ActionModule.java

```diff
@@ -641,6 +642,7 @@ public <Request extends ActionRequest, Response extends ActionResponse> void reg
         actions.register(TransportGetDesiredBalanceAction.TYPE, TransportGetDesiredBalanceAction.class);
         actions.register(TransportDeleteDesiredBalanceAction.TYPE, TransportDeleteDesiredBalanceAction.class);
         actions.register(TransportClusterStatsAction.TYPE, TransportClusterStatsAction.class);
+        actions.register(TransportRemoteClusterStatsAction.TYPE, TransportRemoteClusterStatsAction.class);
```

Comment:
We never execute this action with a local-cluster client, only ever remotely, so no need to register it here.

Reply:
If I remove this I am getting these errors: 
```
Sep 13, 2024 6:33:41 PM com.carrotsearch.randomizedtesting.RandomizedRunner$QueueUncaughtExceptionsHandler uncaughtException
WARNING: Uncaught exception in thread: Thread[elasticsearch-error-rethrower,5,TGRP-ClusterStatsRemoteIT]
java.lang.AssertionError: cluster:monitor/stats/remote
	at __randomizedtesting.SeedInfo.seed([DBE029D8F0FB39F8]:0)
	at org.elasticsearch.transport.InboundAggregator.lambda$new$0(InboundAggregator.java:46)
	at org.elasticsearch.transport.InboundAggregator.initializeRequestState(InboundAggregator.java:197)
	at org.elasticsearch.transport.InboundAggregator.headerReceived(InboundAggregator.java:66)
```

Reply:
Sure, you need to instantiate the `TransportRemoteClusterStatsAction` directly. Doing that in the `TransportClusterStatsAction` constructor makes sense.